obspy/core/utcdatetime.py

```diff
@@ -1282,52 +1271,10 @@ def strftime(self, format):
         See methods :meth:`~datetime.datetime.strftime()` and
         :meth:`~datetime.datetime.strptime()` for more information.
         """
-        # This is an attempt to get consistent behavior across platforms.
-        # See https://bugs.python.org/issue32195
-        # Only ever do anything if there's a real need
-        if self.year < 1000:
-            try:
-                try:
-                    # replace '%Y' depending on results of previous check if
-                    # possible
-                    if not self.__class__._strftime_four_digit_year_works:
-                        format = format.replace("%Y", "%04Y")
-                except AttributeError:
-                    # this is the first time running into a potentially
-                    # problematic call, so check if something needs to be done
-                    # once now
-                    check = self.__class__(900, 1, 1).datetime.strftime('%Y')
-                    self.__class__._strftime_four_digit_year_works = (
-                        len(check) == 4)
-                    # now delegate again
-                    return self.strftime(format)
-            # just a safety net to not break UTCDateTime.strftime if some
-            # unforeseen problems with the above approach appear
-            except Exception:
-                pass
-
+        if sys.platform.startswith("linux"):
```

Comment:
Only needed for linux?

Reply:
To my understanding, yes. This is a glibc implementation thing, `"%0xY"` is only implemented on Linux glibc it seems and Linux might otherwise not put leading zeros, while Mac/Windows seem to put leading zeros nevertheless and rather error out on using `"%0xY"`

Reply:
I'll add those comments pointing to the Python bug reports back in